File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s)!=len(t)):
            return False    
        r={}
        for ch in s:
            if(ch in r):
                r[ch]=r[ch]+1
            else:
                r[ch]=1
        for ch in t:
            if(ch in r):
                r[ch]=r[ch]-1
                if(r[ch]==0):
                    del r[ch]
            else:
                return False

        return True

File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()){
            return false;
        }

        int frequency[26] = {0};
        int m,g;
        for(int i=0; i<s.length(); i++){
            m = (int)(s[i]-'a');
            g = (int)(t[i]-'a');
            frequency[m]++;
            frequency[g]--;
        }
        for(int i=0; i<26; i++){
            if(frequency[i] != 0){
                return false;
            }
        }
        return true;
    }
};